src/components/Pages/SetupPage/SetupPage.tsx

```diff
@@ -67,6 +67,9 @@ const DeployErrorMessage = (props: { message: string }) => (
 )
 
 export const SetupPage = () => {
+  const hasStartedToWriteSomethingInName = useRef(false)
+  const hasStartedToWriteSomethingInEmail = useRef(false)
+  const hasCheckedAgree = useRef(false)
```

Comment:
Why `useRef` and not `useState`?

Reply:
As we don't need the component to trigger an update when these values change, the use of `useRef ` seems fit for this purpose.